### PR TITLE
WIP: store ip allocation identifier on machine creation and free ip o…

### PR DESCRIPTION
**Pull request by erdii**
…n machine deletion

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note

```

### CODE REVIEW

File: pkg/cloudprovider/provider/anexia/provider.go [hunk 1]
```diff
@@ -278,10 +282,19 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, _ *cloudprovidertypes.Prov
 	err = apiClient.VSphere().Provisioning().VM().Deprovision(ctx, status.InstanceID, false)
 	if err != nil {
 		var respErr *anxclient.ResponseError
-		if errors.As(err, &respErr) && respErr.ErrorData.Code == http.StatusNotFound {
-			return true, nil
+		// Only error if the error was not "not found"
+		if !(errors.As(err, &respErr) && respErr.ErrorData.Code == http.StatusNotFound) {
```
Comment (mfranczy) [inline pkg/cloudprovider/provider/anexia/provider.go:281]: why did you add `!` to the condition?

Comment (erdii) [reply pkg/cloudprovider/provider/anexia/provider.go:281]: I inverted it because we have to continue with freeing the ip afterwards.
The `err was 404` path is now a fallthrough to the `free ip` codepath.
